src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/RoundTripIslSpec.groovy

```diff
@@ -0,0 +1,287 @@
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwToDeactivate.dpId).state == SwitchChangeType.ACTIVATED
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.DISCOVERED
+            }.size() == topology.islsForActiveSwitches.size() * 2
+        }
+        isSrcSwDeactivated = false
+
+        then: "Round trip isl is DISCOVERED again"
+        northbound.getLink(roundTripIsl).state == IslChangeType.DISCOVERED
+
+        and: "Round trip status is available again for the given ISL in both directions"
+        [roundTripIsl, roundTripIsl.reversed].each { isl ->
+            assert database.getIslRoundTripStatus(isl) == IslStatus.ACTIVE
+        }
+
+        cleanup:
+        isSrcSwDeactivated && lockKeeper.reviveSwitch(srcSwToDeactivate)
+        isRoundTripRuleDeleted && northbound.installSwitchRules(dstSw.dpId, InstallRulesAction.INSTALL_ROUND_TRIP_LATENCY)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwToDeactivate.dpId).state == SwitchChangeType.ACTIVATED
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.DISCOVERED
+            }.size() == topology.islsForActiveSwitches.size() * 2
+            assert northbound.validateSwitch(dstSw.dpId).rules.missing.empty
+        }
+    }
+
+    @Tidy
+    def "A round trip latency ISL goes down when the switch supports round trip but rule for this purpose is not installed"() {
+        given: "A round trip latency ISL"
+        Isl roundTripIsl
+        def srcSwToDeactivate = topology.activeSwitches.find { sw ->
+            if (sw.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)) {
+                roundTripIsl = topology.getRelatedIsls(sw).find {
+                    it.dstSwitch.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)
+                }
+                roundTripIsl
+            }
+        } ?: assumeTrue("Wasn't able to find a switch with suitable link", false)
+        def dstSw = roundTripIsl.dstSwitch
+
+        when: "Delete ROUND_TRIP_LATENCY_RULE_COOKIE on the dst switch"
+        northbound.deleteSwitchRules(dstSw.dpId, DeleteRulesAction.REMOVE_ROUND_TRIP_LATENCY)
+        def isRoundTripRuleDeleted = true
+        Wrappers.wait(RULES_DELETION_TIME) {
+            assert northbound.validateSwitch(dstSw.dpId).rules.missing.size() == 1
+        }
+
+        and: "Deactivate the src switch"
+        lockKeeper.knockoutSwitch(srcSwToDeactivate)
+        def isSrcSwDeactivated = true
+        // it takes more time to DEACTIVATE a switch via the 'knockoutSwitch' method on the stage env
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwToDeactivate.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        then: "The round trip latency ISL is FAILED"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
```

Comment:
Review all the sleeps. Remove unrequired sleeps, reduce the amount of duplicated time-consuming verifications.

Reply:
fixed everywhere